azkaban-common/src/main/java/azkaban/utils/DependencyTransferManager.java

```diff
@@ -93,45 +100,57 @@ public void downloadAllDependencies(final Set<DependencyFile> deps) {
-      logger.info("Time taken to download all thin archive dependencies in seconds: " + (System.currentTimeMillis() - startTime) / 1000);
+      logger.info("Time taken to download all thin archive dependencies for project {} in {} ms.",
+          projectName, System.currentTimeMillis() - startTime);
     } catch (InterruptedException e) {
       // No point in continuing, let's stop any future downloads and try to interrupt currently running ones.
-      cancelPendingTasks(taskFutures);
+      logger.error("Download interrupted. Cancelling all pending downloads for project {}.", projectName);
+      cancelPendingTasks(taskFutures, projectName);
       throw new DependencyTransferException("Download interrupted.", e);
     } catch (ExecutionException e) {
       // ^^^ see above comment ^^^
-      cancelPendingTasks(taskFutures);
+      logger.error("Download failed. Cancelling all pending downloads for project {}.", projectName);
+      cancelPendingTasks(taskFutures, projectName);
       if (e.getCause() instanceof DependencyTransferException) {
         throw (DependencyTransferException) e.getCause();
       }
       throw new RuntimeException(e.getCause());
+    } catch (TimeoutException e) {
+      logger.error("Timed out while downloading dependencies for project {}.", projectName, e);
+      cancelPendingTasks(taskFutures, projectName);
+      throw new DependencyTransferException("Timed out while downloading dependencies.", e);
     }
   }
 
-  private void downloadDependency(final DependencyFile f) {
+  private void downloadDependency(final DependencyFile f, String projectName) {
     try {
-      downloadDependency(f, 0);
+      downloadDependency(f, 0, projectName);
+      logger.info("Successfully downloaded dependency {} for project {}.", f.getFileName(), projectName);
     } catch (IOException e) {
+      logger.error("Error while downloading dependency {} for project {}.", f.getFileName(), projectName, e);
       throw new DependencyTransferException("Error while downloading dependency " + f.getFileName(), e);
     } catch (HashNotMatchException e) {
+      logger.error("Checksum did not match when downloading dependency {} for project {}.", f.getFileName(), projectName, e);
       throw new DependencyTransferException("Checksum did not match when downloading dependency " + f.getFileName(), e);
     }
   }
 
   /* Cancel all the not-started tasks which are possibly waiting in the queue */
-  private static void cancelPendingTasks(CompletableFuture[] taskFutures) {
-    logger.error("cancelling the pending tasks because one of the downloads failed");
+  private static void cancelPendingTasks(CompletableFuture[] taskFutures, String projectName) {
+    logger.info("cancelling the pending tasks for project {} because one of the downloads failed.", projectName);
+    boolean cancelled = true;
     for (CompletableFuture future : taskFutures) {
-      future.cancel(false);
+      cancelled &= future.cancel(false);
```

Comment:
I'm wondering if we need to log cancelled in the log.
The canceled flag means: 
> The CompletableFuture.cancel method returns a boolean value that indicates whether the cancellation was successful or not. If the CompletableFuture was not previously completed, then it will be cancelled and the method will return true. However, if the CompletableFuture was already completed or cancelled, then the cancel operation will have no effect and the method will return false.

If the cancellation really fails, it will throw an exception

Reply:
Not really, cancellation exception would always throw the difference is whether it is direct CancellationException or ExecutionException wrapping `CancellationException`.
Reference from Java doc:
>      * If not already completed, completes this CompletableFuture with
     * a {@link CancellationException}. Dependent CompletableFutures
     * that have not already completed will also complete
     * exceptionally, with a {@link CompletionException} caused by
     * this {@code CancellationException}.
I don't think it really matters if cancel successfully or not as those tasks are async, I just want to have a log to reveal any issues if we observe in the future that could caused by failure of cancel. But personally I don't think it matters to the upload itself. Just in case this might be useful in the future.

Reply:
Yeah, it also confuses me. If exception would always be thrown, how can you run a for loop here? Wouldn't it break in the middle?

I agree having a log line is nice. I think probably we shouldn't log the "cancelled ? "Successfully" : "Failed to"" in the log line b/c it is not same definition of "Success" or "Fail" we think about when first see this log line. My definition of "Cancel Fail" is: cancellation failed completely but not "it has been completed or cancelled". Maybe that's my own thinking though.

It will confuse our developers who read that log line.

Reply:
This is actually a good discussion. I add one more unit test to verify that.
It turns out cancel() itself won't throw any exception, but it will mark the future with cancellationException. So if we want to get result, it will throw that exception.
And thanks for the log description, I was wrong about the returning value too. Completed future will result in false, I used to think only cancel failed for other reasons will return false. Anyway I removed the boolean value.